proto/internal/uber/cadence/matching/v1/service.proto

```diff
@@ -0,0 +1,214 @@
+
+  // DescribeTaskList returns information about the target tasklist, right now this API returns the
+  // pollers which polled this tasklist in last few minutes.
+  rpc DescribeTaskList(DescribeTaskListRequest) returns (DescribeTaskListResponse);
+
+  // ListTaskListPartitions returns a map of partitionKey and hostAddress for a taskList
+  rpc ListTaskListPartitions(ListTaskListPartitionsRequest) returns (ListTaskListPartitionsResponse);
+}
+
+message PollForDecisionTaskRequest {
+  api.v1.PollForDecisionTaskRequest request = 1;
+  string domain_id = 2;
+  string poller_id = 3;
+  string forwarded_from = 4;
+}
+
+message PollForDecisionTaskResponse {
+  bytes task_token = 1;
+  api.v1.WorkflowExecution workflow_execution = 2;
+  api.v1.WorkflowType workflow_type = 3;
+  int64 previous_started_event_id = 4;
+  int64 started_event_id = 5;
+  int32 attempt = 6;
+  int64 next_event_id = 7;
+  int64 backlog_count_hint = 8;
+  bool sticky_execution_enabled = 9;
+  api.v1.WorkflowQuery query = 10;
+  shared.v1.TransientDecisionInfo decision_info = 11;
+  api.v1.TaskList workflow_execution_task_list = 12;
+  int32 event_store_version = 13;
+  bytes branch_token = 14;
+  google.protobuf.Timestamp scheduled_time = 15 [(gogoproto.stdtime) = true];
+  google.protobuf.Timestamp started_time = 16 [(gogoproto.stdtime) = true];
+  map<string, api.v1.WorkflowQuery> queries = 17;
+}
+
+message PollForActivityTaskRequest {
+  api.v1.PollForActivityTaskRequest request = 1;
+  string domain_id = 2;
+  string poller_id = 3;
+  string forwarded_from = 4;
+}
+
+message PollForActivityTaskResponse {
+  bytes task_token = 1;
+  api.v1.WorkflowExecution workflow_execution = 2;
+  string activity_id = 3;
+  api.v1.ActivityType activity_type = 4;
+  api.v1.Payload input = 5;
+  google.protobuf.Timestamp scheduled_time = 6 [(gogoproto.stdtime) = true];
+  google.protobuf.Timestamp started_time = 7 [(gogoproto.stdtime) = true];
+  google.protobuf.Duration schedule_to_close_timeout = 8 [(gogoproto.stdduration) = true];
+  google.protobuf.Duration start_to_close_timeout = 9 [(gogoproto.stdduration) = true];
+  google.protobuf.Duration heartbeat_timeout = 10 [(gogoproto.stdduration) = true];
+  int32 attempt = 11;
+  google.protobuf.Timestamp scheduled_time_of_this_attempt = 12 [(gogoproto.stdtime) = true];
+  api.v1.Payload heartbeat_details = 13;
+  api.v1.WorkflowType workflow_type = 14;
+  string workflow_domain = 15;
+  api.v1.Header header = 16;
+}
+
+message AddDecisionTaskRequest {
+  string domain_id = 1;
+  api.v1.WorkflowExecution workflow_execution = 2;
+  api.v1.TaskList task_list = 3;
+  int64 schedule_id = 4;
+  google.protobuf.Duration schedule_to_start_timeout = 5 [(gogoproto.stdduration) = true];
+  shared.v1.TaskSource source = 6;
+  string forwarded_from = 7;
+}
+
+message AddDecisionTaskResponse {
+}
+
+message AddActivityTaskRequest {
+  string domain_id = 1;
+  api.v1.WorkflowExecution workflow_execution = 2;
+  string source_domain_id = 3;
```

Comment:
huh.  does this mean domain X can start an activity in domain Y?

Reply:
Not sure about this one's purpose. @meiliang86 do you know what is the use of this field?

Reply:
Not sure either. @yycptt or @yux0 do you know?